cmd/soroban-cli/src/commands/contract/build.rs

```diff
@@ -199,27 +205,62 @@ impl Cmd {
                     return Err(Error::Exit(status));
                 }
 
-                let file = format!("{}.wasm", p.name.replace('-', "_"));
-                let target_file_path = Path::new(target_dir)
-                    .join(&wasm_target)
-                    .join(&self.profile)
-                    .join(&file);
+                self.post_build_steps(&print, &p, target_dir, &wasm_target)?;
+            }
+        }
 
-                self.handle_contract_metadata_args(&target_file_path)?;
+        Ok(())
+    }
 
-                let final_path = if let Some(out_dir) = &self.out_dir {
-                    fs::create_dir_all(out_dir).map_err(Error::CreatingOutDir)?;
-                    let out_file_path = Path::new(out_dir).join(&file);
-                    fs::copy(target_file_path, &out_file_path).map_err(Error::CopyingWasmFile)?;
-                    out_file_path
-                } else {
-                    target_file_path
-                };
+    fn post_build_steps(
+        &self,
+        print: &Print,
+        package: &Package,
+        target_dir: &Utf8PathBuf,
+        wasm_target: &str,
+    ) -> Result<(), Error> {
+        let file = format!("{}.wasm", package.name.replace('-', "_"));
+        let target_file_path = Path::new(target_dir)
+            .join(wasm_target)
+            .join(&self.profile)
+            .join(&file);
+
+        self.handle_contract_metadata_args(&target_file_path)?;
+
+        let final_path = if let Some(out_dir) = &self.out_dir {
+            fs::create_dir_all(out_dir).map_err(Error::CreatingOutDir)?;
+            let out_file_path = Path::new(out_dir).join(&file);
+            fs::copy(&target_file_path, &out_file_path).map_err(Error::CopyingWasmFile)?;
+            out_file_path
+        } else {
+            target_file_path.clone()
+        };
 
-                Self::print_build_summary(&print, &final_path)?;
+        if self.optimize {
+            #[cfg(feature = "additional-libs")]
+            {
+                use crate::wasm::Args as WasmArgs;
+                let optimized_file_path = final_path.with_extension("optimized.wasm");
+                let wasm_args = WasmArgs {
+                    wasm: final_path.clone(),
+                };
+                wasm_args.optimize(&optimized_file_path)?;
+                Self::print_build_summary_with_optimization(
+                    print,
+                    &final_path,
+                    &optimized_file_path,
+                )?;
             }
+            #[cfg(not(feature = "additional-libs"))]
+            {
+                print.warnln("Optimization not available. Install with \"additional-libs\" feature to use --optimize.");
```

Comment:
One last thing, this should be a hard error imo. The command was asked to optimize and the feature isn't available, so someone should have to rerun the command without --optimize.